File: 1160 - Find Words That Can Be Formed by Characters.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int countCharacters(String[] words, String chars) {
        Map<Character, Integer> count = new HashMap<>();
        for (Character c : chars.toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }
        int res = 0;
        for (String word : words) {
            Map<Character, Integer> Tcount = new HashMap<>();
            boolean good = true;
            for (Character c : word.toCharArray()) {
                Tcount.put(c, Tcount.getOrDefault(c, 0) + 1);
            }
            for (Character c : Tcount.keySet()) {
                if (!count.containsKey(c)) {
                    good = false;
                } else if (count.get(c) < Tcount.get(c)) {
                    good = false;
                }
            }
            if (good) {
                res += word.length();
            }
        }
        return res;
    }
}

File: 1160 - Find Words That Can Be Formed by Characters.js
/**
 * @param {string[]} words
 * @param {string} chars
 * @return {number}
 */
var countCharacters = function (words, chars) {
    let count = new Map();
    for (let c of chars) {
        if (count.has(c)) {
            count.set(c, count.get(c) + 1);
        } else {
            count.set(c, 1);
        }
    }
    let res = 0;
    for (let word of words) {
        let Tcount = new Map();
        for (let c of word) {
            if (Tcount.has(c)) {
                Tcount.set(c, Tcount.get(c) + 1);
            } else {
                Tcount.set(c, 1);
            }
        }
        let good = true;
        for (let k of Tcount.keys()) {
            if (!count.has(k)) {
                good = false;
            } else if (count.get(k) < Tcount.get(k)) {
                good = false;
            }
        }
        if (good) {
            res += word.length;
        }
    }
    return res;
};